pyresample/geometry.py

```diff
@@ -269,9 +269,19 @@ def get_boundary_lonlats(self):
         return (SimpleBoundary(s1_lon.squeeze(), s2_lon.squeeze(), s3_lon.squeeze(), s4_lon.squeeze()),
                 SimpleBoundary(s1_lat.squeeze(), s2_lat.squeeze(), s3_lat.squeeze(), s4_lat.squeeze()))
 
-    def get_bbox_lonlats(self) -> tuple:
+    def get_bbox_lonlats(self, force_clockwise: bool = True) -> tuple:
         """Return the bounding box lons and lats.
 
+        Args:
+            force_clockwise:
+                Perform minimal checks and reordering of coordinates to ensure
+                that the returned coordinates follow a clockwise direction.
+                This is important for compatibility with
+                :class:`pyresample.spherical.AreaBoundary`. This is required in
+                cases where data is not oriented in the traditional way where
+                the first element of data (row 0, col 0) is the north-west
```

Comment:
An ascending pass and descending pass of a well behaved sensor (clockwise edges) have opposite first element orientations: north-west and south-east (not necessarily respectively, depending on how you look at it).
So I don't think north-west should be mentioned here, since the flip of lons and lats doesn't actually change the position of the first corner of the swath, right?

Reply:
What if I mentioned north-west and south-east?

Reply:
I think north-east and south-west cases can also happen (I'm thinking with Sentinel1 where I've seen this happen)

Reply:
So you're saying that Sentinel1 data has either north-east or south-west as its first corner and the path followed by the existing logic still produces a clockwise series of coordinates? I suppose anything's possible. I'll see if I can remove the mention of north-west and be clearer about what I'm saying.

Reply:
yes exactly. The sentinel 1 data we use comes flipped horizontally or vertically depending whether the pass is ascending or descending...